File: impl/app/android/app/src/main/java/com/ak/contexta/ui/components/BottomNavBar.kt
package com.ak.contexta.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AutoStories
import androidx.compose.material.icons.outlined.Home
import androidx.compose.material.icons.outlined.MenuBook
import androidx.compose.material.icons.outlined.Settings
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.ak.contexta.navigation.Screen
import com.ak.contexta.ui.theme.Hairline
import com.ak.contexta.ui.theme.Muted
import com.ak.contexta.ui.theme.Primary

enum class BottomNavTab(val route: String, val label: String, val icon: ImageVector) {
    Home(Screen.Home.route, "首页", Icons.Outlined.Home),
    Vocabulary(Screen.Vocabulary.route, "生词", Icons.Outlined.MenuBook),
    Reference(Screen.Reference.route, "参考", Icons.Outlined.AutoStories),
    Settings(Screen.Settings.route, "设置", Icons.Outlined.Settings)
}

@Composable
fun BottomNavBar(
    selectedTab: BottomNavTab,
    onTabSelected: (BottomNavTab) -> Unit,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier.fillMaxWidth()) {
        // 1px hairline top border
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(Hairline)
        )
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(MaterialTheme.colorScheme.surface)
                .heightIn(min = 44.dp)
                .padding(vertical = 6.dp),
            horizontalArrangement = Arrangement.SpaceEvenly,
            verticalAlignment = Alignment.CenterVertically
        ) {
            BottomNavTab.entries.forEach { tab ->
                val isSelected = tab == selectedTab
                val color = if (isSelected) Primary else Muted
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    modifier = Modifier
                        .clickable { onTabSelected(tab) }
                        .padding(horizontal = 16.dp)
                ) {
                    Icon(
                        imageVector = tab.icon,
                        contentDescription = tab.label,
                        tint = color,
                        modifier = Modifier.height(24.dp)
                    )
                    Text(
                        text = tab.label,
                        style = MaterialTheme.typography.titleSmall,
                        color = color
                    )
                }
            }
        }
    }
}

File: impl/app/flutter/lib/core/components/bottom_nav_bar.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/app_type.dart';

/// 底部导航 tab（对照 Kotlin BottomNavTab：Home/Vocabulary/Reference/
/// Settings 四个一级页面；Reading/AddWord/Onboarding 不在底栏）。
enum BottomNavTab {
  home(route: 'home', label: '首页', icon: Icons.home_outlined),
  vocabulary(route: 'vocabulary', label: '生词', icon: Icons.menu_book_outlined),
  reference(route: 'reference', label: '参考', icon: Icons.auto_stories_outlined),
  settings(route: 'settings', label: '设置', icon: Icons.settings_outlined);

  const BottomNavTab({
    required this.route,
    required this.label,
    required this.icon,
  });

  final String route;
  final String label;
  final IconData icon;
}

/// 底部导航栏：顶部 1dp Hairline 分隔线 + 4 tab；选中 = Primary 珊瑚、
/// 未选中 = Muted；标签 titleSmall。对照 Kotlin ui/components/BottomNavBar.kt。
class BottomNavBar extends StatelessWidget {
  const BottomNavBar({
    super.key,
    required this.selectedTab,
    required this.onTabSelected,
  });

  final BottomNavTab selectedTab;
  final ValueChanged<BottomNavTab> onTabSelected;

  @override
  Widget build(BuildContext context) {
    return Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        const Divider(
          height: 1,
          thickness: 1,
          color: AppColors.hairline,
        ),
        Material(
          color: AppColors.surfaceCard,
          child: Padding(
            padding: const EdgeInsets.symmetric(vertical: 6),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: [
                for (final tab in BottomNavTab.values)
                  _TabItem(
                    tab: tab,
                    selected: tab == selectedTab,
                    onTap: () => onTabSelected(tab),
                  ),
              ],
            ),
          ),
        ),
      ],
    );
  }
}

class _TabItem extends StatelessWidget {
  const _TabItem({
    required this.tab,
    required this.selected,
    required this.onTap,
  });

  final BottomNavTab tab;
  final bool selected;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final color = selected ? AppColors.primary : AppColors.muted;
    return InkWell(
      onTap: onTap,
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(tab.icon, size: 24, color: color),
            Text(
              tab.label,
              style: AppType.textTheme.titleSmall?.copyWith(color: color),
            ),
          ],
        ),
      ),
    );
  }
}
